src/seedu/addressbook/data/person/ReadOnlyPerson.java

```diff
@@ -83,4 +84,17 @@ default String getAsTextHidePrivate() {
         }
         return builder.toString();
     }
+    
+    /**
+     * Returns a concatenated version of the printable strings of each object.
+     */
+   default String getPrintableString(Printable... printables){
+       String printableString = "";
+       for (Printable printable : printables) {
+           printableString += printable.getPrintableString();
+           printableString += SEPARATOR;
+       }
+       printableString = printableString.substring(0, -2);
```

Comment:
Perhaps you can look into `StringJoiner` in the java api to help in concatenating strings?